src/analysis/processing/qgsalgorithmextractbylocation.cpp

```diff
@@ -452,6 +452,11 @@ QVariantMap QgsSelectByLocationAlgorithm::processAlgorithm( const QVariantMap &p
     {
       // When subsetting or removing we only need to check already selected features
       std::unique_ptr< QgsVectorLayerSelectedFeatureSource > selectLayerSelected( new QgsVectorLayerSelectedFeatureSource( selectLayer ) );
+
+      mTargetCrs = selectLayerSelected->sourceCrs();
+      mTargetFeatureCount = selectLayerSelected->featureCount();
+      mIntersectFeatureCount = intersectSource->featureCount();
```

Comment:
Shall we not move this to `prepareAlgorithm`? because `processAlgorithm` can be run in a different thread and thus this variables are needed in the others case block `process` calls

Reply:
@troopa81 , select by location is not thread safe (as reflected by the flags), so I'm not touching it.

Reply:
Right! :+1: But what happen for other `AddToSelection` and `SetSelection` method ? the above variables won't be initialized when calling process.

Reply:
@troopa81 , well spotted, addressed.